microservices/update_user_ext_source.py

```diff
@@ -0,0 +1,282 @@
+from adapters.AdaptersManager import AdaptersManager
+from satosa.micro_services.base import ResponseMicroService
+from typing import List, Union, Any
+from utils.ConfigStore import ConfigStore
+
+import threading
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class UpdateUserExtSource(ResponseMicroService):
+    def __init__(self, config, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.DEBUG_PREFIX = "update_user_ext_source: "  # ??
+        self.NAMESPACE = "namespace"
+        self.UES_ATTR_NMS = "urn:perun:ues:attribute-def:def"
+        self.DEFAULT_CONFIG = {
+            'sourceIdPAttributeKey': 'sourceIdPEntityID',
+            'userIdentifiers': ['eduPersonUniqueId',
+                                'eduPersonPrincipalName',
+                                'eduPersonTargetedID',
+                                'nameid',
+                                'uid']
+        }
+
+        self.__adapters_manager = AdaptersManager()
+        self.__attr_map = config['attrMap']
+        self.__append_only_attrs = []
+        self.__array_to_str_conversion = []
+        if config['arrayToStringConversion']:
+            self.__array_to_str_conversion = config['arrayToStringConversion']
+        if config['arrayToStringConversion']:
+            self.__append_only_attrs = config['appendOnlyAttrs']
+
+    def process(self, context, data):
+        data_to_conversion = {
+            'attributes': data['Attributes'],

```

Comment:
```suggestion
            'attributes': data.attributes,

```
in SATOSA it's `data.attributes` (see https://github.com/IdentityPython/SATOSA/blob/1de6a360d52369e01a40d7d0400af20735d7e899/src/satosa/internal.py#L114)